relay-server/src/services/project_cache.rs

```diff
@@ -469,10 +472,14 @@ impl ProjectSource {
 
         #[cfg(feature = "processing")]
         if let Some(redis_source) = self.redis_source {
-            let state_fetch_result =
-                tokio::task::spawn_blocking(move || redis_source.get_config(project_key))
-                    .await
-                    .map_err(|_| ())?;
+            let redis_permit = self.redis_semaphore.clone().acquire_owned().await?;
+            let state_fetch_result = tokio::task::spawn_blocking(move || {
+                let ret = redis_source.get_config(project_key);
+                drop(redis_permit);
+                ret
+            })
+            .await
+            .map_err(|_| ())?;
```

Comment:
```suggestion
            {
                 let _permit = self.redis_semaphore.acquire().await?;
                 let state_fetch_result = tokio::task::spawn_blocking(move || {
                     redis_source.get_config(project_key)
                 })
                 .await
                .map_err(|_| ())?;
            }
```

We don't need to move the guard into the task here, we want to limit concurrently active calls to `spawn_blocking`, which means we can just drop the permit after the call returned.